main.tf

```
@@ -22,7 +23,7 @@ resource "aws_route53_zone" "default" {
     "$${stage}", module.this.stage),
     "$${id}", module.this.id),
     "$${attributes}", join(module.this.delimiter, module.this.attributes)),
-    "$${parent_zone_name}", coalesce(join("", data.aws_route53_zone.parent_zone.*.name), var.parent_zone_name)),
+    "$${parent_zone_name}", coalesce(join("", data.aws_route53_zone.parent_zone.*.name), local.parent_zone_name)),
```

Also replace `var.zone_name` on L19 with `local.zone_name`

```suggestion
    "$${parent_zone_name}", coalesce(join("", data.aws_route53_zone.parent_zone.*.name), var.parent_zone_name, "none")),
```